drivers/gles3/storage/mesh_storage.cpp

```diff
@@ -1502,7 +1502,7 @@ void MeshStorage::multimesh_set_visible_instances(RID p_multimesh, int p_visible
 
 	if (multimesh->data_cache.size()) {
 		//there is a data cache..
-		_multimesh_mark_all_dirty(multimesh, false, true);
+		_multimesh_mark_all_dirty(multimesh, true, true);
```

Comment:
I would replace this with a call to ``_multimesh_mark_dirty()`` and then call it for each region that needs to be updated. that way we only have to update the regions that actually need to be updated.